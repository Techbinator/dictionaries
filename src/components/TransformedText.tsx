import React from "react";
import { ITranslations } from "../store/Dictionaries.type";

interface ITransformedText {
  domains: string[];
  text: string;
  translations: ITranslations;
}
export default ({ domains, text, translations }: ITransformedText) => {
  if (!domains.includes(text.toString())) {
    return <td>{text}</td>;
  }
  const translationUUID = Object.keys(translations).find(
    key => translations[key].domain.toString() == text.toString()
  );
  if (!translationUUID) {
    return <td>{text}</td>;
  }
  const translation = translations[translationUUID];

  return (
    <td>
      <span style={{ color: "red" }}>{translation.domain}</span>/
      <span style={{ color: "green" }}>{translation.range}</span>
    </td>
  );
};
