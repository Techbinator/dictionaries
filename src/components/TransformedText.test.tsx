import React from "react";
import TransformedText from "./TransformedText";
import { shallow } from "enzyme";
import contextMock from "../helpers/__mocks__/context.mock";

describe("Header component", () => {
  it("should render snapshot with default text", () => {
    const wrapper = shallow(
      <TransformedText
        domains={["test"]}
        text="text"
        translations={contextMock.translations}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should render snapshot with transformed text", () => {
    const wrapper = shallow(
      <TransformedText
        domains={["Turquoise"]}
        text="Turquoise"
        translations={contextMock.translations}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
