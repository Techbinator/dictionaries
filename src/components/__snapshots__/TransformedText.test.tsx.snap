// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render snapshot with default text 1`] = `
<td>
  text
</td>
`;

exports[`Header component should render snapshot with transformed text 1`] = `
<td>
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    Turquoise
  </span>
  /
  <span
    style={
      Object {
        "color": "green",
      }
    }
  >
    newRange
  </span>
</td>
`;
